search/searcher/search_phrase.go

```diff
@@ -55,6 +56,13 @@ func (s *PhraseSearcher) Size() int {
 
 func NewMultiPhraseSearcher(indexReader search.Reader, terms [][]string, field string, scorer search.Scorer,
 	options search.SearcherOptions) (*PhraseSearcher, error) {
+	return NewSloppyMultiPhraseSearcher(indexReader, terms, field, 0, scorer, options)
+}
+
+// NewSloppyMultiPhraseSearcher create a multi-phrase searcher which tolerates a specified "sloppyness"
+// the value of the slop parameter restricts the distance between the terms
+func NewSloppyMultiPhraseSearcher(indexReader search.Reader, terms [][]string, field string, slop int,
```

Comment:
One question here is why you decided to have a new method?  Looking at the code, we only call this in one place outside of tests (MultiPhraseQuery Searcher method).

Since Bluge is still developer preview, I'd prefer we change the API rather than add extra methods.  But I'm open to being persuaded if you have compelling reasons here.

Reply:
I had three options here, 
1. Make Slop a required parameter, breaking the existing API
2. Add another method make the first one retains it's current behavior with default "slop" value
3. Start using pattern like [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis)

I chose the second one because it has the smallest blast radius, didn't break the existing API and didn't require adopting a pattern with isn't used in Bluge.

Happy to change this to something better, which options would you prefer? or a different suggestion?